components/cd-disclosure/cd-disclosure.js

```diff
@@ -0,0 +1,81 @@
+(function (Drupal) {
+  'use strict';
+
+  /**
+   * @see https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/
+   */
+  Drupal.behaviors.cdDisclosure = {
+    idPrefix: 'cd-disclosure--',
+
+    attach: function (context, settings) {
+      // Initialize toggable dropdown.
+      this.initializeDisclosures();
+    },
+
+    /**
+     * Set up all Disclosure widgets found in the DOM.
+     *
+     * For our purposes, the `disclosure` element is the wrapped content itself
+     * and the button to toggle it is generated automatically by this script.
+     */
+    initializeDisclosures: function () {
+      let disclosures = document.querySelectorAll('.cd-disclosure');
+
+      // All behavior is controlled by data attributes.
+      disclosures.forEach((disclosure) => {
+        // Set a random ID if none is found.
+        let disclosureId = disclosure.getAttribute('id') || this.idPrefix + Math.floor(Math.random(1) * 100000000);
+
+        // Apply ID if it wasn't there already.
+        if (!disclosure.getAttribute('id')) {
+          disclosure.setAttribute('id', disclosureId);
+        }
+
+        // Determine where to place the button.
+        let positionAfter = disclosure.dataset.cdDisclosureInsert === 'after' || false;
+
+        // Determine whether content is shown by default. We consider this the
+        // edge-case so omitting the attribute assumes it is hidden.
+        let contentVisible = disclosure.dataset.cdDisclosureShowByDefault === 'true' || false;
+
+        // Extract button label
+        let buttonText = disclosure.dataset.cdDisclosureLabel || 'Toggle content';
+
+        // Create the toggler button.
+        let disclosureButton = document.createElement('button');
+        disclosureButton.setAttribute('aria-controls', disclosureId);
+        disclosureButton.setAttribute('aria-expanded', contentVisible);
+        disclosureButton.innerText = buttonText;
+        disclosureButton.classList.add('cd-button');
```

Comment:
This is missing some classes to be able to style the button (for example to add an arrow icon via `::after` for example) something like `cd-disclosure-toggler`?

Reply:
Added a new attribute and docs to README. It still outputs `cd-button` by default, but now you can zero it out with an empty attribute, or add space-separated classes to your heart's desire.